src/ValidateModelStateAttribute.cs

```diff
@@ -2,23 +2,64 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Linq;
+using System.Text;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.Logging;
 
 namespace Microsoft.DotNet.HelixPoolProvider
 {
-    public class ValidateModelStateAttribute : ActionFilterAttribute
+    public class ValidateModelStateAttribute : TypeFilterAttribute
     {
-        public ValidateModelStateAttribute()
+        public ValidateModelStateAttribute() : base(typeof(ValidateModelStateImpl))
         {
             Order = int.MaxValue;
         }
 
-        public override void OnActionExecuting(ActionExecutingContext context)
+        private class ValidateModelStateImpl : IActionFilter
         {
-            if (!context.ModelState.IsValid)
+            private readonly ILogger<ValidateModelStateAttribute> _logger;
+
+            public ValidateModelStateImpl(ILogger<ValidateModelStateAttribute> logger)
+            {
+                _logger = logger;
+            }
+
+            public void OnActionExecuting(ActionExecutingContext context)
+            {
+                if (!context.ModelState.IsValid)
```

Comment:
Suggest to invert `if`s to reduce nesting. Makes the code significantly easier to read, maitain and test